packages/adminrouter/extra/src/README.md

```diff
@@ -893,3 +899,835 @@ expressions/statements:
+  end
+  ```
+  and the endpoint configuration will now look like this:
+  ```
+  location /schmetterlingdb/ {
+      access_by_lua_block {
+        auth.access_schmetterlingdb_endpoint();
+      }
+      proxy_pass http://schmetterlingdb/;
+  }
+  ```
+
+* with the mock endpoint working and authentication in place, we can add
+  authentication tests. Depending on the flavour where our SchmetterlingDB is
+  present (open/ee/both), this must be done in either one of these or both:
+  * for EE, add endpoint path and RID to either `acl_endpoints` or
+    `authed_endpoints` lists from `test-harness/tests/ee/test_master.py`. See
+    the comments in the file for more details.
+  * for Open, add your endpoint to `authed_endpoints` list in
+    `test-harness/tests/open/test_master.py`
+* even though `/schmetterlingdb/stats/` does not require authentication it still
+  needs to be tested. This should be done using generic tests. Depending on the
+  AR flavour which `SchmetterlingDB` is designed to work with, we need
+  the following:
+  ```
+  - tests:
+      is_unauthed_access_permitted:
+        locations:
+          - /schmetterlingdb/stats/foo/bar
+    type:
+      - master
+  ```
+  in one of the files:
+  * EE: `test-harness/tests/ee/test_generic.config.yml`
+  * Open: `test-harness/tests/open/test_generic.config.yml`
+  * both: `test-harness/tests/test_generic.config.yml`
+
+  See section describing [generic tests syntax](#yaml-file-syntax) for more
+  details. In all the following examples, there is going to be implicit assumption
+  that the user already knows the file/place where generic tests belong to.
+
+* next - we need to make sure that our endpoints are proxing to correct upstreams.
+  This should be done using `is_upstream_correct` test:
+  ```
+  - tests:
+      is_upstream_correct:
+        test_paths:
+          - /schmetterlingdb/stats/foo/bar
+          - /schmetterlingdb/foo/bar
+        upstream: http://127.0.0.1:12345
+    type:
+      - master
+  ```
+  We need to test *both* locations as each one of them is a separate block in
+  AR configuration, which may point to a different upstream.
+
+* next - we need to make sure that `/schmetterlingdb` request is properly
+  redirected to `/schmetterlingdb/`. It is important to point out that all
+  redirects should be done using `307 Temporary Redirect` as:
+  * 301,302 redirects do not preserve request method (clients change POST to
+    GET)
+  * 308 redirect is not supported by all clients
+
+  The test is as follows:
+  ```
+  - tests:
+      is_endpoint_redirecting_properly:
+        locations:
+          - path: /schmetterlingdb
+            code: 307
+    type:
+      - master
+  ```
+* next - each request needs to set standardised set of headers to upstream.
+  This list includes:
+  * `Host`
+  * `X-Real-IP`
+  * `X-Forwarded-For`
+  * `X-Forwarded-Proto`
+
+  On top of that, our SchmetterlingDB does not neet JWT, so it is going to be
```

Comment:
> does not neet JWT

- `s/neet/need/`
- I think here we really should say "DC/OS authentication token" instead of JWT. "JWT" is an implementation detail, and in the long term we will have various tokens in the system some of which will be JWTs, but only one class of which will be DC/OS authentication tokens.